addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -267,11 +270,56 @@ private void localFileReflectionAttack(HttpMessage msg) {
                             .setEvidence(matcher.group())
                             .setMessage(msg)
                             .raise();
+                    alertRaised = true;
                 }
                 if (isStop()) {
                     return;
                 }
             }
+
+            // Now if no issue is found yet, then we replace the values one at a time. Do this for a
+            // fixed number of Elements, depending on the strength at which the rule is used.
+            if (!alertRaised) {
+                // Remove original xml header
+                Matcher headerMatcher = xmlHeaderPattern.matcher(originalRequestBody);
+                String headerlessRequestBody = headerMatcher.replaceAll("");
+                int totalTagCount = getTotalTagCount(headerlessRequestBody);
+                int maxValuesChanged = 0;
+
+                if (this.getAttackStrength() == AttackStrength.MEDIUM) {
+                    maxValuesChanged = 72 / LOCAL_FILE_TARGETS.length;
+                } else if (this.getAttackStrength() == AttackStrength.HIGH) {
+                    maxValuesChanged = 144 / LOCAL_FILE_TARGETS.length;
+                } else if (this.getAttackStrength() == AttackStrength.INSANE) {
+                    maxValuesChanged = totalTagCount;
+                }
+
+                for (int tagIdx = 1;
+                        ((tagIdx <= totalTagCount) && (tagIdx <= maxValuesChanged));
+                        tagIdx++) {
+                    requestBody = createTagSpecificLfrPayload(headerlessRequestBody, tagIdx);
+                    for (int idx = 0; idx < LOCAL_FILE_TARGETS.length; idx++) {
+                        String localFile = LOCAL_FILE_TARGETS[idx];
+                        payload = MessageFormat.format(requestBody, localFile);
+                        msg.setRequestBody(payload);
+                        sendAndReceive(msg);
+                        String response = msg.getResponseBody().toString();
+                        Matcher matcher = LOCAL_FILE_PATTERNS[idx].matcher(response);
```

Comment:
This intermediate storage probably isn't necessary if response is never used further, it can just:
`Matcher matcher = LOCAL_FILE_PATTERNS[idx].matcher(msg.getResponseBody().toString());` directly.

Reply:
Should I squash the commit for this with the previous one?

Reply:
@kingthorin Done